### DISCUSSION TITLE
Why is `.git/lfs/tmp` 17GB after `git lfs prune --recent`?

### DISCUSSION THREAD

**Discussion by zwimer** [git-lfs/git-lfs#5912] [Q&A] [answered]
I have a `git` as a submodule, call it `SUBM`; this submodule has `lfs` 2 objects in it totaling about `16GB`.

After running a `git lfs prune --recent`, I found that:
- `.git/modules/SubModuleName/lfs/objects` is `16 GB`
- `.git/modules/SubModuleName/lfs/tmp` is `17 GB`.

As far as I can tell, this means I likely have 3 copies of my LFS data on my local machine for a single commit (`HEAD`) since no other branches have this LFS data, nor do any previous commits have any any differing LFS data.

Is there anyway I can remove the data in `lfs/tmp` without re-pulling the LFS data from the remote (bandwidth is limited)?

**Comment by chrisd8088** [comment, marked as answer]
Hey, thanks for the question!  I think if you run any Git LFS command while your current working directory is within your sub-project, it will perform its usual cleanup of the Git LFS temporary directory for that sub-project.  The normal expiry time for files in that directory is [one hour](https://github.com/git-lfs/git-lfs/blob/3990c7ab4fa6c99540af3f73f33da47a72df08af/fs/cleanup.go#L75-L79), and the cleanup functions are [run](https://github.com/git-lfs/git-lfs/blob/3990c7ab4fa6c99540af3f73f33da47a72df08af/git-lfs.go#L35) after every Git LFS command, even the non-destructive ones like `git lfs ls-files`.

However, the key thing here is to run such a command within the sub-project's working tree.  You can see why if you run `git lfs env | grep ^TempDir` in both the parent super-project and the sub-project.  For example:

```
$ git init parent
$ cd parent
$ git submodule add https://github.com/example/test  ## this is empty, but we don't care here
$ git lfs env | grep ^TempDir
TempDir=/path/to/parent/.git/lfs/tmp
$ cd test
$ git lfs env | grep ^TempDir
TempDir=/path/to/parent/.git/modules/test/lfs/tmp
```

So the temporary directory which the Git LFS client will clear depends on whether you're in or out of the sub-project's working tree.  Given that, I think that if you `cd SubModuleName` in your main project's working tree and then run really any Git LFS command like `git lfs ls-files` or even just `git lfs env`, that should clear out any stale files in the sub-project's Git LFS temporary directory.
